lib/conjur/base.rb

```diff
@@ -351,8 +351,12 @@ def mtime
       end
 
       def refresh_token
+        # There's a race condition here in which the file could be updated
+        # after we read the mtime but before we read the file contents. So to be
+        # conservative, use the oldest possible mtime.
+        mtime = self.mtime
```

Comment:
But I suppose it doesn't hurt to be more conservative; the worst that could happen is the file will get read twice.